core/src/transaction.rs

```diff
@@ -74,53 +67,146 @@ impl rlp::Decodable for UnverifiedTransaction {
     }
 }
 
-impl rlp::Encodable for UnverifiedTransaction {
+impl TempTransaction {
+    /// Append object with a signature into RLP stream
+    fn rlp_append_sealed_transaction(&self, s: &mut RlpStream) {
+        s.begin_list(6);
+        s.append(&self.unsigned.seq);
+        s.append(&self.unsigned.fee);
+        s.append(&self.unsigned.network_id);
+        s.append(&self.unsigned.action);
+        s.append(&self.sig);
+        s.append(&self.signer_public);
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Verified(TempTransaction);
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Unverified(pub TempTransaction);
+
+impl rlp::Encodable for Verified {
     fn rlp_append(&self, s: &mut RlpStream) {
-        self.rlp_append_sealed_transaction(s)
+        self.0.rlp_append(s);
+    }
+}
+
+impl rlp::Decodable for Verified {
```

Comment:
This path may be violate the invariant that `VerifiedTransaction` was always verified. If no usage is found then remove it or convert it first to `Unverified` then utilize `try_into`